contracts/MigrationContractStaging.cdc

```diff
@@ -0,0 +1,362 @@
+/// This contract is intended for use in the Cadence 1.0 contract migration across the Flow network.
+///
+/// In preparation for this milestone, your contract will NEED to be updated! Once you've updated your code for
+/// Cadence 1.0, you MUST stage your contract code in this contract so that the update can be executed as a part of the
+/// network-wide state migration.
+///
+/// To stage your contract update:
+/// 1. create a Host & save in your contract-hosting account
+/// 2. call stageContract() passing a reference to you Host, the contract name, and the hex-encoded Cadence code
+///
+/// This can be done in a single transaction! For more code context, see https://github.com/onflow/contract-updater
+///
+access(all) contract MigrationContractStaging {
+
+    // Path constants
+    //
+    access(self) let delimiter: String
+    access(self) let capsulePathPrefix: String
+    access(all) let HostStoragePath: StoragePath
+    access(all) let AdminStoragePath: StoragePath
+    /// Maps contract addresses to an array of staged contract names
+    access(self) let stagedContracts: {Address: [String]}
+    /// The block height at which updates can no no longer be staged. If nil, updates can be staged indefinitely until
+    /// the cutoff value is set.
+    access(all) var stagingCutoff: UInt64?
+
+    /// Event emitted when a contract's code is staged
+    /// status == true - insert | staged == false - replace | staged == nil - remove
+    /// NOTE: Does not guarantee that the contract code is valid Cadence
+    access(all) event StagingStatusUpdated(
+        capsuleUUID: UInt64,
+        address: Address,
+        codeHash: [UInt8],
+        contract: String,
+        status: Bool?
```

Comment:
personally i think a string that just uses the words "insert", "replace", "remove" is better here then this.

Reply:
Good point! Updated to `action` field with values `"stage", "replace", and "unstage"